File: staGledas.Service/Database/KlubFilmovaClanovi.cs
using System;

namespace staGledas.Service.Database
{
    public class KlubFilmovaClanovi
    {
        public int Id { get; set; }
        public int KlubId { get; set; }
        public int KorisnikId { get; set; }
        public string Uloga { get; set; } = "member"; // owner, admin, member
        public DateTime DatumPridruzivanja { get; set; }
        public virtual KlubFilmova? Klub { get; set; }
        public virtual Korisnici? Korisnik { get; set; }
    }
}

File: staGledas-UI/stagledas_mobile/lib/models/klub_filmova_clanovi.dart
import 'package:json_annotation/json_annotation.dart';
import 'korisnik.dart';

part 'klub_filmova_clanovi.g.dart';

@JsonSerializable()
class KlubFilmovaClanovi {
  int? id;
  int? klubId;
  int? korisnikId;
  String? uloga;
  DateTime? datumPridruzivanja;
  Korisnik? korisnik;

  KlubFilmovaClanovi({
    this.id,
    this.klubId,
    this.korisnikId,
    this.uloga,
    this.datumPridruzivanja,
    this.korisnik,
  });

  bool get isOwner => uloga == 'owner';
  bool get isMember => uloga == 'member';

  factory KlubFilmovaClanovi.fromJson(Map<String, dynamic> json) =>
      _$KlubFilmovaClanoviFromJson(json);

  Map<String, dynamic> toJson() => _$KlubFilmovaClanoviToJson(this);
}
